trestle/tasks/xlsx_to_oscal_component_definition.py

```diff
@@ -0,0 +1,671 @@
+from trestle.oscal.common import ResponsibleParty
+from trestle.oscal.common import ResponsibleRole
+from trestle.oscal.common import Role
+from trestle.oscal.component import ComponentDefinition
+from trestle.oscal.component import ControlImplementation
+from trestle.oscal.component import DefinedComponent
+from trestle.oscal.component import ImplementedRequirement
+from trestle.oscal.component import SetParameter
+from trestle.oscal.component import Statement
+from trestle.tasks.base_task import TaskBase
+from trestle.tasks.base_task import TaskOutcome
+from trestle.utils.oscal_helper import CatalogHelper
+from trestle.utils.parameter_helper import ParameterHelper
+
+logger = logging.getLogger(__name__)
+
+t_component_name = str
+t_control = str
+t_description = str
+t_goal_id = int
+t_goal_name_id = str
+t_goal_remarks = str
+t_goal_text = str
+t_goal_version = str
+t_guidelines = str
+t_name = str
+t_parameter_usage = str
+t_parameter_value = str
+t_parameter_values = str
+t_row = int
+t_statements = List[str]
+t_tokens = List[str]
+t_type = str
+t_uuid_str = str
+t_values = str
+t_work_sheet = Worksheet
+
+t_controls = Dict[t_control, t_statements]
+
+
+class XlsxToOscalComponentDefinition(TaskBase):
+    """
+    Task to create OSCAL ComponentDefinition json.
+
+    Attributes:
+        name: Name of the task.
+    """
+
+    name = 'xlsx-to-oscal-component-definition'
+
+    def __init__(self, config_object: Optional[configparser.SectionProxy]) -> None:
+        """
+        Initialize trestle task xlsx-to-oscal-component-definition.
+
+        Args:
+            config_object: Config section associated with the task.
+        """
+        super().__init__(config_object)
+        self._timestamp = datetime.datetime.utcnow().replace(microsecond=0).replace(tzinfo=datetime.timezone.utc
+                                                                                    ).isoformat()
+
+    def set_timestamp(self, timestamp) -> None:
+        """Set the timestamp."""
+        self._timestamp = timestamp
+
+    def print_info(self) -> None:
+        """Print the help string."""
+        logger.info(f'Help information for {self.name} task.')
+        logger.info('')
+        logger.info(
+            'Purpose: From spread sheet and catalog produce Open Security Controls Assessment Language (OSCAL) component definition file.'
+        )
+        logger.info('')
+        logger.info('Configuration flags sit under [task.xlsx-to-oscal-component-definition]:')
+        logger.info(
+            '  catalog-file      = (required) the path of the OSCAL catalog file, for example '
+            + self._get_catalog_title() + '.'
+        )
+        logger.info(
+            '  spread-sheet-file = (required) the path of the spread sheet file, containing data for production of component definition.'
+        )
+        logger.info('  work-sheet-name   = (required) the name of the work sheet in the spread sheet file.')
```

Comment:
Two issues here:
1) Goals are a vendor specific construct. I think we may need another formulation (checks / rules / etc.)
2) Here the spreadsheet is effectively static including locked to users format. I would have though we could do this by heading lookup or similar (it might be worth looking at pandas for parsing excel files as it does this well).

Reply:
1. I left this alone for now. If you feel strongly, I can try to externalize to config file.
2. Fixed. Using column name now.